cmd/bee/cmd/start.go

```diff
@@ -169,7 +169,7 @@ func (c *command) initStartCmd() (err error) {
 
 			interruptChannel := make(chan struct{})
 
-			b, err := node.NewBee(interruptChannel, c.config.GetString(optionNameP2PAddr), signerConfig.publicKey, signerConfig.signer, networkID, logger, signerConfig.libp2pPrivateKey, signerConfig.pssPrivateKey, &node.Options{
+			b, err := node.NewBee(interruptChannel, sysInterruptChannel, c.config.GetString(optionNameP2PAddr), signerConfig.publicKey, signerConfig.signer, networkID, logger, signerConfig.libp2pPrivateKey, signerConfig.pssPrivateKey, &node.Options{
```

Comment:
So I remember in [this PR](https://github.com/ethersphere/bee/pull/3127), we decided not to pass the sysInterruptChannel down stream. Although in this case it seems to be okay as we are only using it during the NewBee function. This can be easily misused.

One way would be to wrap the checking of sysInterruptChan in a closure which can be used by NewBee. If we plan to remove this in future, we should create an issue in the backlog for this.

Reply:
you are right, this is a fix that would live only until v1.9, added the issue here: https://github.com/ethersphere/bee/issues/3254